src/mscorlib/shared/System/SpanHelpers.Char.cs

```diff
@@ -167,6 +168,90 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
             }
         }
 
+        public static unsafe int LastIndexOf(ref char searchSpace, char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            fixed (char* pChars = &searchSpace)
+            {
+                char* pCh = pChars + length;
+                char* pEndCh = pChars;
+
+#if !netstandard11
+                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
+                {
+                    const int elementsPerByte = sizeof(ushort) / sizeof(byte);
+                    length = (((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) / elementsPerByte) + 1;
+                }
+            SequentialScan:
+#endif
+                while (length >= 4)
+                {
+                    length -= 4;
+                    pCh -= 4;
+
+                    if (*(pCh + 3) == value)
+                        goto Found3;
+                    if (*(pCh + 2) == value)
+                        goto Found2;
+                    if (*(pCh + 1) == value)
+                        goto Found1;
+                    if (*pCh == value)
+                        goto Found;
+                }
+
+                while (length > 0)
+                {
+                    length -= 1;
+                    pCh -= 1;
+
+                    if (*pCh == value)
+                        goto Found;
+                }
+#if !netstandard11
+                // We get past SequentialScan only if IsHardwareAccelerated is true. However, we still have the redundant check to allow
+                // the JIT to see that the code is unreachable and eliminate it when the platform does not have hardware accelerated.
+                if (Vector.IsHardwareAccelerated && pCh > pEndCh)
+                {
+                    length = (int)((pCh - pEndCh) & ~(Vector<ushort>.Count - 1));
+
+                    // Get comparison Vector
+                    Vector<ushort> vComparison = new Vector<ushort>(value);
+
+                    while (length > Vector<ushort>.Count - 1)
+                    {
+                        char* pStart = pCh - Vector<ushort>.Count;
+                        // Using Unsafe.ReadUnaligned instead of Read since it isn't gauranteed that pStart is vector aligned
```

Comment:
Why is this not same as in IndexOf?

```
     // Using Unsafe.Read instead of ReadUnaligned since the search space is pinned and pCh is always vector aligned
     Debug.Assert(((int)pCh & (Unsafe.SizeOf<Vector<ushort>>() - 1)) == 0);
```

Reply:
@ahsonkhan - I thought we discussed this last night, and I showed you why `pStart` is guaranteed to be aligned.

Reply:
Yes, `char* pStart = pCh - Vector<ushort>.Count + 1;` is always aligned, but in the current implementation, that causes test failures. For correctness, I had to change it to `char* pStart = pCh - Vector<ushort>.Count;` which isn't aligned.

I will investigate how to fix the test failures.

As part of this change, I also had to change the sequential scan (and the goto logic) for correctness. We decrement `pch` at the start of the loop before comparing with value rather than doing it at the end of the loop.